apps/web/src/services/project/components/ProjectFormModal.vue

```diff
@@ -74,43 +82,87 @@ const {
     },
 });
 
-/* Event */
-const confirm = async () => {
-    if (state.loading) return;
-    if (!isAllValid.value) return;
-
+/* Api */
+const createProject = async (): Promise<ProjectModel|undefined> => {
     state.loading = true;
-
     try {
         const params: ProjectCreateParameters = {
             name: name.value?.trim() as string,
-            project_type: 'PRIVATE', // TODO: project_type
+            project_type: state.selectedAccess,
         };
         if (props.projectGroupId) params.project_group_id = props.projectGroupId;
-        const project = await projectPageStore.createProject(params);
-        if (project) {
-            state.proxyVisible = false;
-            emit('confirm');
-        }
+        return await projectPageStore.createProject(params);
     } catch (e) {
         ErrorHandler.handleError(e);
+        return undefined;
     } finally {
         state.loading = false;
     }
 };
+const updateProject = async (): Promise<ProjectModel|undefined> => {
+    try {
+        const params: ProjectUpdateParameters = {
+            project_id: props.project.project_id || router.currentRoute.params.id,
+            name: name.value?.trim() as string,
+        };
+        return await projectPageStore.updateProject(params);
+    } catch (e) {
+        return undefined;
+    }
+};
+const updateProjectType = async (): Promise<ProjectModel|undefined> => {
+    try {
+        const params: ProjectUpdateProjectTypeParameters = {
+            project_id: props.project.project_id || router.currentRoute.params.id,
+            project_type: state.selectedAccess,
+        };
+        return await projectPageStore.updateProjectType(params);
+    } catch (e) {
+        ErrorHandler.handleRequestError(e, i18n.t('PROJECT.DETAIL.ALT_E_UPDATE_PROJECT'));
+        return undefined;
+    }
+};
+
+/* Event */
+const confirm = async () => {
+    if (state.loading) return;
+    if (!isAllValid.value) return;
+
+    if (props.project) { // update project
+        const updatedProject1 = await updateProject();
+        const updatedProject2 = await updateProjectType();
+        if (updatedProject1 && updatedProject2 || updatedProject2) {
+            await projectStore.sync(updatedProject2);
+            emit('confirm', updatedProject2);
+        } else if (updatedProject1) {
+            await projectStore.sync(updatedProject1);
+            emit('confirm', updatedProject1);
+        }
```

Comment:
description: Code for handling cases where **all fail** or **only one fails**.

Reply:
🤔